src/prog_models/sim_result.py

```diff
@@ -165,17 +173,35 @@ def pop(self, index : int = -1):
             return self.__data.pop(index)
         return self.fcn(x)
 
-    def remove(self, target) -> None:
+    def remove(self, d = None, t = None, s = None) -> None:
         """Remove an element
-
+         
         Args:
-            target: Value of element to be removed.
-        """
-        target_index = self.data.index(target)
-        self.times.pop(target_index)
-        self.states.pop(target_index)
-        if self.__data is not None:
-            self.__data.remove(target)
+            d: Data value to be removed.
+            t: Time value to be removed.
+            s: State value to be removed.
+        """ 
+        if (not [i for i in (d,t,s) if i is not None or None]) or (sum([i is None for i in (d, t, s)]) != 2):
```

Comment:
```suggestion
        if sum([i is None for i in (d, t, s)]) != 2:
```
I don't think we need the first part. Just make sure there are exactly 2 Nones

Reply:
Tested, this is true. Making change